core/src/test/scala/unit/kafka/server/DeleteTopicsRequestWithDeletionDisabledTest.scala

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import kafka.network.SocketServer
+import kafka.utils._
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.apache.kafka.common.requests.{DeleteTopicsRequest, DeleteTopicsResponse}
+import org.junit.Assert._
+import org.junit.Test
+
+import scala.collection.JavaConverters._
+
+class DeleteTopicsRequestWithDeletionDisabledTest extends BaseRequestTest {
```

Comment:
Should we have an integration test for AdminClient as well?

Reply:
I can include another class overwriting the prop ``delete.topic.enable`` to test it. Not a big deal. I think it can be useful, what do you say?

Reply:
maybe we can add another test to this class. we can borrow some code from [AdminClientIntegrationTest.scala](https://github.com/apache/kafka/blob/trunk/core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala#L211)

Reply:
That's what I was saying @omkreddy. It'd be write another class extending ``IntegrationTestHarness`` disabling the property and including this test.
But maybe adding a simple unit test in ``KafkaAdminClientTest``as @hachikuji suggested it'd be fine (btw, we don't have unit tests to remove topics in that class).
So?

Reply:
@mmolimar I'm ok either way. I'd probably lean toward a unit test in `KafkaAdminClientTest` since integration tests are more expensive for the build time. And yeah, there does seem to be a testing gap for the error handling of this API.

Reply:
Added! ;-)